cypress/support/step_definitions/getterSetterCalls.js

```diff
@@ -16,25 +16,509 @@
+          'Unable to find the firebolt object in the runtime environment variable';
+      }
+
+      cy.then(() => {
+        // Failing the test when fireboltCall object not there
+        if (!fireboltCallObject) {
+          fireLog.assert(false, fireboltCallObjectErrorMessage);
+        } else {
+          let setMethod = UTILS.resolveRecursiveValues(fireboltCallObject.setMethod);
+          let setParams;
+
+          // Extracting the parameter from the fireboltCall object
+          if (
+            fireboltCallObject.setParams &&
+            typeof fireboltCallObject.setParams === CONSTANTS.TYPE_OBJECT
+          ) {
+            setParams = fireboltCallObject.setParams;
+
+            // Iterating through the object and invoke if it is a function
+            for (const key in setParams) {
+              if (typeof setParams[key] === CONSTANTS.TYPE_FUNCTION) {
+                setParams[key] = setParams[key]();
+              }
+            }
+          } else {
+            setParams = { value: UTILS.resolveRecursiveValues(fireboltCallObject.setParams) };
+          }
+
+          const context = {};
+          const expected = invalidValue ? CONSTANTS.ERROR : CONSTANTS.RESULT;
+          let action = CONSTANTS.ACTION_CORE.toLowerCase();
+
+          // Splitting the method name if it contains an underscore and using the first part to determine the action that decides sdk.
+          if (setMethod && setMethod.includes('_')) {
+            action = setMethod.split('_')[0];
+            setMethod = setMethod.split('_')[1];
+          }
+
+          const additionalParams = {
+            method: setMethod,
+            params: setParams,
+            context: context,
+            action: action,
+            expected: expected,
+            appId: Cypress.env(CONSTANTS.FIRST_PARTY_APPID),
+          };
+
+          fireLog.info(
+            'Call from 1st party App, method: ' +
+              setMethod +
+              ' params: ' +
+              JSON.stringify(setParams)
+          );
+          cy.sendMessageToPlatformOrApp(CONSTANTS.PLATFORM, additionalParams);
+        }
+      });
+    } else {
+      fireLog.assert(false, `${sdk} SDK not Supported`);
+    }
+  }
+);
+
+/**
+ * @module getterSetterCalls
+ * @function Given '(.+)' invokes the '(.+)' get API(?: '(.+)')
+ * @description Sending a message to platform or app to get a value
+ * @param {String} appId - app identifier.
+ * @param {String} sdk - sdk name.
+ * @param {String} fireboltCallKey - key name passed to look for firebolt call object in fireboltCallData.
+ * @example
+ * Given '1st party app' invokes the 'Firebolt' get API 'CLOSEDCAPTION_SETTINGS'
+ * Given '3rd party app' invokes the 'Firebolt' get API 'CLOSEDCAPTION_SETTINGS'
+ * Given '3rd party app' invokes the 'Firebolt' get API
+ * Given 'test_app' invokes the 'Firebolt' get API 'CLOSEDCAPTION_SETTINGS'
+ */
+Given(/'(.+)' invokes the '(.+)' get API(?: '(.+)')?$/, async (appId, sdk, fireboltCallKey) => {
```

Comment:
With the exception of the `appId` logic this code seems very similar to the previous one. Can we extract common functionality into a helper method to be used by both?

Reply:
In upcoming enhancement ticket observed we are going to modify the glue and removing some fields, After removing some of the lines we need to remove. Is it to handle thins in new ticket?